packages/core-utils/src/type-utils/events.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * Provides type safety for decoupled events mechanics.
+ *
+ * @example
+ * interface MyComponentEvents {
+ *   onEvent1: { a: string, b: number },
+ *   onEvent2: null
```

Comment:
Since the keys are event names, they likely won't include `on`.

```suggestion
 *   event1: { a: string, b: number },
 *   event2: null
```